migrations/versions/0425_unique_service_name_2.py

```diff
@@ -0,0 +1,37 @@
+"""
+
+Revision ID: 0425_unique_service_name_2
+Revises: 0424_n_history_created_at
+Create Date: 2023-09-01 10:10:19.011060
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = "0425_unique_service_name_2"
+down_revision = "0424_n_history_created_at"
+
+
+def upgrade():
+    op.execute("UPDATE services SET normalised_service_name = email_from WHERE normalised_service_name IS NULL")
+    op.execute("UPDATE services_history SET normalised_service_name = email_from WHERE normalised_service_name IS NULL")
+    op.alter_column("services", "normalised_service_name", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("services_history", "normalised_service_name", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("services", "email_from", existing_type=sa.VARCHAR(), nullable=True)
+    op.alter_column("services_history", "email_from", existing_type=sa.VARCHAR(), nullable=True)
+    op.drop_constraint("services_email_from_key", "services", type_="unique")
```

Comment:
This is a bit thorny. We may want to keep this constraint for now. if we drop it, we might get into a situation where we can't downgrade (if the email_from becomes duplicated in some way).

Reply:
we need to drop this here as we're going to stop updating this value from the rest of the front end as part of our renaming step.

given that normalised_service_name already has a unique_constraint, by setting `email_from` to be the value from normalised_service_name in the downgrade script we should be covered i believe.

i might remove the where clause from the downgrade just to be safe.

Reply:
Okay that makes sense. Looks good